src/dash-table/derived/table/index.tsx

```diff
@@ -26,25 +28,53 @@ const handleSetFilter = (
     setState({workFilter: {map, value: filter_query}, rawFilterQuery});
 };
 
-function propsAndMapFn(propsFn: () => ControlledTableProps, setFilter: any) {
+function propsAndMapFn(
+    propsFn: () => ControlledTableProps,
+    setFilter: any,
+    toggleFilterOptions: (column: IColumn) => IColumn
+) {
     const props = propsFn();
 
-    return R.merge(props, {map: props.workFilter.map, setFilter});
+    return R.merge(props, {
+        map: props.workFilter.map,
+        setFilter,
+        toggleFilterOptions
+    });
 }
 
 export default (propsFn: () => ControlledTableProps) => {
     const setFilter = memoizeOne((setProps: SetProps, setState: SetState) =>
         handleSetFilter.bind(undefined, setProps, setState)
     );
 
+    const toggleFilterOptions = memoizeOne(
+        (setProps: SetProps, columns: IColumn[]) => (column: IColumn) => {
+            const newColumns = [...columns];
+            const iColumn = columns.indexOf(column);
+
+            const newColumn = {...newColumns[iColumn]};
+            newColumn.filter_options =
+                newColumn.filter_options === FilterCase.Insensitive
+                    ? FilterCase.Sensitive
+                    : FilterCase.Insensitive;
+
+            newColumns.splice(iColumn, 1, newColumn);
+
+            setProps({columns: newColumns});
+
+            return newColumn;
```

Comment:
update the table's columns in the shallowest way possible (minimize cache busting)